packages/client/src/provider/in-memory-provider/in-memory-provider.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  FlagValueType,
+  GeneralError,
+  JsonValue,
+  Logger,
+  OpenFeatureError,
+  OpenFeatureEventEmitter,
+  ProviderEvents,
+  ResolutionDetails,
+  StandardResolutionReasons,
+  TypeMismatchError,
+  ProviderStatus,
+  Flag,
+  FlagConfiguration,
+  VariantFoundError,
+} from '@openfeature/shared';
+import { Provider } from '../provider';
+
+/**
+ * A simple OpenFeature provider intended for demos and as a test stub.
+ */
+export class InMemoryProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
+  public readonly runsOn = 'client';
+  status: ProviderStatus = ProviderStatus.NOT_READY;
+  readonly metadata = {
+    name: 'in-memory',
+  } as const;
+  private _flagConfiguration: FlagConfiguration;
+  private _flagEvalCache: Record<string, ResolutionDetails<number | string | boolean | JsonValue>> = {};
+
+  constructor(flagConfiguration: FlagConfiguration = {}) {
+    this._flagConfiguration = { ...flagConfiguration };
+  }
+
+  async initialize(context?: EvaluationContext | undefined): Promise<void> {
+    this.status = ProviderStatus.READY;
```

Comment:
I tried this before and ended up doing it this way :sweat_smile: 

Because of how `lookUpFlagValue` is implemented the tests suite broke.
The provider state is `stale`, and it gives `cached` as reason.